oauth2_http/javatests/com/google/auth/oauth2/IdentityPoolCredentialsTest.java

```diff
@@ -1060,7 +1036,135 @@ public void serialize() throws IOException, ClassNotFoundException {
+    certificateMap.put("use_default_certificate_config", true);
+    Map<String, Object> credentialSourceMap = new HashMap<>();
+    credentialSourceMap.put("certificate", certificateMap);
+    IdentityPoolCredentialSource credentialSource =
+        new IdentityPoolCredentialSource(credentialSourceMap);
+    MockExternalAccountCredentialsTransportFactory mockTransportFactory =
+        new MockExternalAccountCredentialsTransportFactory();
+
+    // Build credentials with the custom provider.
+    IdentityPoolCredentials credentials =
+        IdentityPoolCredentials.newBuilder()
+            .setX509Provider(x509Provider)
+            .setHttpTransportFactory(mockTransportFactory)
+            .setAudience("test-audience")
+            .setSubjectTokenType("test-token-type")
+            .setCredentialSource(credentialSource)
+            .build();
+
+    // Verify successful creation and correct internal setup.
+    assertNotNull("Credentials should be successfully created", credentials);
+    assertTrue(
+        "Subject token supplier should be for certificates",
+        credentials.getIdentityPoolSubjectTokenSupplier()
+            instanceof CertificateIdentityPoolSubjectTokenSupplier);
+    assertEquals(
+        "Metrics header should indicate certificate source",
+        IdentityPoolCredentials.CERTIFICATE_METRICS_HEADER_VALUE,
+        credentials.getCredentialSourceType());
+
+    // Verify the custom provider methods were called during build.
+    verify(x509Provider).getKeyStore();
+    verify(x509Provider).getCertificatePath();
+  }
+
+  @Test
+  public void build_withDefaultCertificate_throwsOnTransportInitFailure() {
+    // Setup credential source to use default certificate config.
+    Map<String, Object> certificateMap = new HashMap<>();
+    certificateMap.put("use_default_certificate_config", true);
+    Map<String, Object> credentialSourceMap = new HashMap<>();
+    credentialSourceMap.put("certificate", certificateMap);
+    IdentityPoolCredentialSource credentialSource =
+        new IdentityPoolCredentialSource(credentialSourceMap);
+
+    // Expect RuntimeException during build due to mTLS setup failure because the certificate file
+    // doesn't exist.
+    IdentityPoolCredentials.Builder builder =
+        IdentityPoolCredentials.newBuilder()
+            .setHttpTransportFactory(new MockExternalAccountCredentialsTransportFactory())
+            .setAudience("test-audience")
+            .setSubjectTokenType("test-token-type")
+            .setCredentialSource(credentialSource);
+    RuntimeException exception = assertThrows(RuntimeException.class, builder::build);
+
+    assertEquals(
+        "Failed to initialize IdentityPoolCredentials from certificate source due to an I/O error.",
+        exception.getMessage());
+  }
+
+  @Test
+  public void build_withCustomProvider_throwsOnGetKeyStore()
+      throws IOException, KeyStoreException, CertificateException, NoSuchAlgorithmException {
```

Comment:
Same here. I'm not 100% sure why it's supposed to fail. Is this to mimic the an IO error in the keystore?